app/Models/Concerns/HasSlug.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Models\Concerns;
+
+use Illuminate\Support\Str;
+use Illuminate\Database\Eloquent\Model;
+
+trait HasSlug
```

Comment:
Do we want to consider something like https://github.com/spatie/laravel-sluggable. Which has built-in support for slug collisions?

Reply:
Will pull in on separate PR.